android/test_app/app/src/main/cpp/pytorch_testapp_jni.cpp

```diff
@@ -0,0 +1,74 @@
+#include <android/log.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <cassert>
+#include <cmath>
+#include <vector>
+#define ALOGI(...) \
+  __android_log_print(ANDROID_LOG_INFO, "PyTorchTestAppJni", __VA_ARGS__)
+#define ALOGE(...) \
+  __android_log_print(ANDROID_LOG_ERROR, "PyTorchTestAppJni", __VA_ARGS__)
+
+#include "jni.h"
+
+#include <ATen/NativeFunctions.h>
+#include <torch/script.h>
+
+namespace pytorch_testapp_jni {
+namespace {
+
+template <typename T>
+void log(const char* m, T t) {
+  std::ostringstream os;
+  os << t << std::endl;
+  ALOGI("%s %s", m, os.str().c_str());
+}
+
+struct JITCallGuard {
```

Comment:
I feel it's probably helpful to copy the comment in your README here as well as people might copy/paste the code snippet without reading readme, e.g.:
```
To load torchscript model for mobile we need some special setup which is placed in `struct JITCallGuard` in this example. It may change in future, you can track the latest changes keeping an eye in our [pytorch android jni code]([https://github.com/pytorch/pytorch/blob/master/android/pytorch_android/src/main/cpp/pytorch_jni_jit.cpp#L28)
```

also, maybe change "need some special setup" to a bit more context, e.g. "need set these guards because mobile build doesn't support features like autograd for smaller build size"